docs/rfcs/20210223_external_connectivity_in_k8s.md

```diff
@@ -0,0 +1,195 @@
+each Kubernetes node. Operator's responsibility is to use that
+assigned port in initial script that mangles Redpanda configuration.
+
+## Impact
+
+The Kubernetes worker nodes must open port per Redpanda cluster
+which might be an additional attack vector.
+
+# Motivation
+## Why are we doing this?
+
+Redpanda can be easily managed in multiple environments, but vast
+range of users plan to manage their workload in Kubernetes
+environment. Saying that, teams would like to not introduce
+clients application into their Kubernetes clusters. The Kafka API
+requires that each broker will be reachable at any given time.
+Canonical way of exposing webservers is to create Ingress resources
+that can be reconciled into cloud provider load balancer. Another
+way of exposing different kind of workloads is to expose their TCP
+port using again cloud provider load balancer. Ont the other hand
+the TCP port can be exposed on each Kubernetes worker node using
+Service type Node Port. All the mentioned methods doesn't guarantee
+that each Redpanda node will be individually accessible. For that
+reasons the Kubernetes model of exposing workloads outside of the
+k8s cluster doesn't provide predictable way of reaching individual
+broker in Redpanda cluster.
+
+The operator should accommodate dynamic nature of any cloud
+where nodes can disappear or be rescheduled. The reporting of the
+brokers addresses should be as quick as possible.
+
+## What use cases does it support?
+
+Exposing each Redpanda node to the internet.
+
+## What is the expected outcome?
+
+The Redpanda operator will make sure that from the network standpoint
+brokers will be reachable. Users need to managed security of the
+exposed Kubernetes worker nodes e.g. `aws security groups`.
+
+# Guide-level explanation
+## How do we teach this?
+
+The Kafka API requires that each broker will be reachable from
+the client side. To expose Redpanda nodes, but not add additional
+latency to the communication each Redpanda process will have assigned
+unique host port. In case of cloud providers the host VM need to be
+publicly reachable and have opened port. The operator would be
+responsible to orchestrate Kubernetes to achieve that.
+
+## Dictionary
+
+- Kubernetes Service type NodePort with empty selector that gives
+  unused port that is available across all k8s worker nodes
+
+# Reference-level explanation
+
+( This is the technical portion of the RFC )
+
+## Interaction with other features
+
+- The kafka multi listener needs to be set up in each Redpanda
+  node, so that internal connectivity can stay as is. For external
+  connectivity the dedicated port would be opened.
+- The statefulset must bootstrap Redpanda cluster with or without
+  external connectivity.
+- The user intention defined in cluster.redpanda.vectorized.io custom
+  resource should be reconciled into set of ready to use address with
+  ports that can be reach from the internet. The status field will
```

Comment:
Yes. It would include ports.

Examples:
```
# kubectl get clusters.redpanda.vectorized.io -o yaml
.....
  status:
    nodes:
      external:
      - 3.8.157.0:30792
      - 18.134.10.120:30792
      - 18.132.36.174:30792
      internal:
      - external-connectivity-0.external-connectivity.redpanda-system.svc.cluster.local.
      - external-connectivity-1.external-connectivity.redpanda-system.svc.cluster.local.
      - external-connectivity-2.external-connectivity.redpanda-system.svc.cluster.local.
    replicas: 3
    upgrading: false
```